### PR TITLE
Add ImageFeatureExtractionMixin

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR adds a new `ImageFeatureExtractionMixin` to implement the common functionality needed for images (conversion to PIL Image / NumPy array /normalize/ resize) in a framework agnostic way. While it only adds support for torch (not tf) tensors as input, support for TF tensors is easy to add in the design and will be done when we have a TF model with a vision modality.

Along the way, this PR adds a new `is_vision_available` check (depends only on PIL for now, but we can add other dependencies later on if we fill we need them. It could for instance check for torchvision when torch is installed) and the "dummy" vision objects.

I will work on adding tests tomorrow, but the general design can already be reviewed to check if it has everything needed.
cc @NielsRogge

### CODE REVIEW

File: tests/test_image_utils.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+    from transformers import ImageFeatureExtractionMixin
+
+
+def get_random_image(height, width):
+    random_array = np.random.randint(0, 256, (height, width, 3), dtype=np.uint8)
+    return PIL.Image.fromarray(random_array)
+
+
+@require_vision
+class Speech2TextFeatureExtractionTester(unittest.TestCase):
```
Comment (LysandreJik) [inline tests/test_image_utils.py]: This should probably rename to something unrelated to S2T :)

File: src/transformers/testing_utils.py [hunk 2]
```diff
@@ -229,23 +230,17 @@ def require_torch_scatter(test_case):
...
 def require_torchaudio(test_case):
     """
-    Decorator marking a test that requires torchaudio.
-
-    These tests are skipped when torchaudio isn't installed.
-
+    Decorator marking a test that requires torchaudio. These tests are skipped when torchaudio isn't installed.
     """
-    if not is_torchaudio_available:
+    if not is_torchaudio_available():
```
Comment (LysandreJik) [inline src/transformers/testing_utils.py:235]: nice catch

**Review by LysandreJik — APPROVED** [review]
LGTM!

File: tests/test_image_utils.py (hunk 1 above)
Comment (sgugger) [reply tests/test_image_utils.py]: Way ahead of you :-p

**Pull request merged**